src/lib.rs

```diff
@@ -1,3 +1,114 @@
+//! Floem is responsible for making a view active. for handling drag and drop mechanics
+//! It's a View's responsibility to handle events and decide whether they're handled.
+//!
+//! ## State management  
+//!
+//! You will probably want your view components to have some state. You should place any state that affects
+//! you view inside a signal so that you can react to updates and update the `View`. Signals are reactive values that can be read from and written to.
+//!
+//! ### Use state to update your view
+//!
+//! To affect the layout and rendering of your component, you will need to send a state update to your component with [Id::update_state](id::Id::update_state)
+//! and then call [UpdateCx::request_layout](context::UpdateCx::request_layout) to request a layout which will cause a repaint.
+//!
+//! ## Local state
+//!
+//! To share state between components child and sibling components, you can simply pass down a signal to your children. Here's a contrived example:
+//!
+//!```rust,no_run
+//!
+//! struct Parent<V> {
+//!     id: Id,
+//!     text: ReadSignal<String>,
+//!     child: V,
+//! }
+//!
+//! // Creates a new parent view with the given child.
+//! fn parent<V>(new_child: impl FnOnce(ReadSignal<String>) -> V) -> Parent<impl View>
+//! where
+//!     V: View + 'static,
+//! {
+//!     let text = create_rw_signal(cx.scope, "World!".to_string());
+//!     // share the signal between the two children
+//!     let (id, child) = AppContext::new_id_with_child(stack(|| (text_input(text)), new_child(text.read_only()));
+//!     Parent { id, text, child }
+//! }
+//!
+//! impl<V> View for Parent<V>
+//! where
+//!     V: View,
+//! {
+//! // implementation omitted for brevity
+//! }
+//!
+//! struct Child {
+//!     id: Id,
+//!     label: Label,
+//! }
+//!
+//! /// Creates a new child view with the given state (a read only signal)
+//! fn child(text: ReadSignal<String>) -> Child {
+//!     let (id, label) = AppContext::new_id_with_child(|| label(move || format!("Hello, {}", text.get()));
+//!     Child { id, label }
+//! }
+//!
+//! impl View for Child {
+//!   // implementation omitted for brevity
+//! }
+//!
+//! // Usage
+//! fn main() {
+//!     floem::launch(parent(child));
+//! }
+//!
+//! ```
+//!
+//! ### Global state
+//!
+//! Global state can be implemented using Leptos' [provide_context](leptos_reactive::provide_context) and [use_context](leptos_reactive::use_context).
+//!
+//! # Styling
+//! Floem sizing and positioning layout system is based on the flexbox model using Taffy as the layout engine.
+//! Styles are divided into two parts:
+//! [`ComputedStyle`]: A style with definite values for most fields.  
+//!
+//! [`Style`]: A style with [`StyleValue`]s for the fields, where `Unset` falls back to the relevant
+//! field in the [`ComputedStyle`] and `Base` falls back to the underlying [`Style`] or the
+//! [`ComputedStyle`].
+//!
+//!
+//! ## Render loop and update lifecycle
+//! TBD: event -> update -> layout -> paint.
```

Comment:
@dzhou121 or @jrmoulton would love an overview here. I can spend more time going through the code but would love to learn how it's supposed to work at a high level from you.

Reply:
The cycle is correct.

After an event comes in (e.g. the user clicked the mouse, pressed a key etc), the event will be propagated from the root view to the children. The parent will decide whether sending the event to the child(ren) based on the logic in the ```event``` method in the parent View. There's also event listeners that users can use to respond to events the users choose. The event propagation is stopped whenever a child or an event listener returns ```false``` on the event handling. 

During the event handling, there could be state changes through the reactive system. E.g., on the counter example, when you click increment, it updates the counter and because the label listens to the change (probably a link to the doc of the reactive system?), the label will update the text it presents. The reactive system can't directly manipulate the state of the label because the ```AppState``` owns all the views. And instead, it will send the update to a message queue. After the event propagation is done, Floem will process all the update messages in the queue, and it can manipulate the state of a particular view through the ```update``` method. 

The update of states on the Views could cause some of them need a new layout recalculation, because the size might have changed etc. The ```layout``` method is called from the root view to re-layout the views that have requested a layout call. The layout call is to change the layout properties at ```Taffy```, so after the layout call is done, ```compute_layout``` is called to calculate the sizes and positions of each view. 

And in the end, ```paint``` is called to render all the views to the screen.

Reply:
Amazing description! Thank you